docs/configuring-a-resource.md

```diff
@@ -342,6 +342,9 @@ type Reference struct {
 }
 ```
 
+> [!Warning]
+> Please note the `Type` field will be deprecated soon, use `TerraformName` instead.
```

Comment:
nit: We may give some motivation for the deprecation.
```suggestion
> Please note the `Reference.Type` field will be deprecated soon, use `Reference.TerraformName` instead. `TerraformName` is a more stable and less error prone API compared to `Type` because it automatically accounts for the configuration changes affecting the cross-resource reference target's kind name, group or version.
```

Reply:
Done, thank you!